addons/account/migrations/10.0.1.1/post-migration.py

```diff
@@ -116,3 +117,58 @@ def migrate(env, version):
     openupgrade.load_data(
         cr, 'account', 'migrations/10.0.1.1/noupdate_changes.xml',
     )
+
+    # gather non-zero bank statement lines that have journal entries,
+    # but no linked payments, meaning they are reconciled, but no
+    # payment was created.
+    openupgrade.logged_query(cr, '''
+        select l.id
+        from account_bank_statement_line l
+        join account_move m
+            on m.statement_line_id = l.id
+        join account_move_line ml
+            on ml.move_id = m.id
+        left join account_payment ap
+            on ml.payment_id = ap.id
+        where abs(l.amount) > 0.0001
+        group by l.id
+        having (
+            sum(case when ap.id is not null then 1 else 0 end) = 0
+            and sum(case when ml.id is not null then 1 else 0 end) > 0
+        )
+    ''')
+    statement_lines = env['account.bank.statement.line'].browse([
+        row[0] for row in cr.fetchall()])
+
+    # create payments for these
+    for st_line in statement_lines:
+        total = st_line.amount
+        journal = st_line.journal_id
+        currency = journal.currency_id or st_line.company_id.currency_id
+        payment_methods = (total > 0) and \
```

Comment:
As a rule of thumb, use if else in cases like these (`journal.inbound_payment_method_ids if total > 0 else outbound_payment_method_ids`) to prevent `journal.outbound_payment_method_ids` kicking in when `journal.inbound_payment_method_ids` is falsy even when total < 0.

Reply:
Nicely spotted, thanks.